### PR TITLE
Close serial port connection before flashing firmware

### LINKED ISSUE #684: Firmware update fails if Serial Monitor is open
## Describe the bug

The Arduino IDE includes a "**WiFi101 / WiFiNINA Firmware Updater**"" tool that can be used to update the firmware on the WiFi radio module of Arduino boards such as the [MKR WiFi 1010](https://store.arduino.cc/arduino-mkr-wifi-1010) and [Nano 33 IoT](https://store.arduino.cc/arduino-nano-33-iot).

If the Serial Monitor is open when a firmware update is attempted, it fails.

## To Reproduce

1. Connect an Arduino board that has a u-blox NINA-W102 module to your computer via the USB cable.
1. Select the board and port in the IDE.
1. Open the Arduino IDE Serial Monitor (**Tools > Serial Monitor**).
1. Select **Tools > WiFi101 / WiFiNINA Firmware Updater** from the Arduino IDE menus.
1. Select the board in the **Firmware Updater** dialog.
1. Click the <kbd>Check Updates</kbd> button.
1. Click the <kbd>Install</kbd> button
   (it doesn’t matter which firmware version you flash).
1. Wait for the update process to finish.

The process fails without any indication of the cause

> ⚠️ Installation failed. Please try again.

## Expected behavior

Ideally, the IDE's Firmware Updater tool would automatically close the serial port as needed, reopening it in Serial Monitor once done, similar to how it is handled during the Upload process.

If that is not feasible, it should check for open Serial Monitor at the start of the firmware update process and clearly communicate to the user that they must close it. This is the approach taken by the classic IDE's firmware updater tool:

Clipboard01

## Desktop

- OS: Windows 10
- Version: 2.0.0-rc2
Date: 2021-12-13T10:09:39.122Z
CLI Version: 0.20.2 [13783819]

## Additional context

I think it could be a common occurrence because most often people are motivated to update their firmware after seeing the outdated firmware message that is printed to Serial Monitor by the "CheckFirmwareVersion" example of the WiFiNINA library or by the ArduinoIoTCloud library.

### PR DESCRIPTION
**Pull request by AlbyIanna**
Fixes #684

### CODE REVIEW

**Comment by AlbyIanna** [comment]
This PR fixes the issue (you can flash the firmware when the serial monitor/plotter is open), but a little problem remains: when you flash the firmware to a board, the sketch on the board is replaced with the code necessary to do the update, so re-opening the serial port is pretty much pointless. I wonder if this impacts the experience 
cc: @per1234 @ubidefeo

**Comment by per1234** [comment]
>re-opening the serial port is pretty much pointless. I wonder if this impacts the experience

I don't think so. The reason is that if I am working with the Serial Monitor before doing a firmware update, I am likely to want to go back to working with Serial Monitor again after. So it is convenient for Serial Monitor to remain open, even though there will be no output until I have uploaded my sketch again.

The firmware updater already displays a warning about the sketch being overwritten, so the user shouldn't be surprised to find that no output is arriving in the Serial Monitor. However, they would find it confusing to see the "Not connected" banner persist in Serial Monitor after finishing a firmware update, since that could give the impression that something is wrong.

**Review by per1234 — APPROVED** [review]
This fixes https://github.com/arduino/arduino-ide/issues/684 for me. Thanks @AlbyIanna!

File: arduino-ide-extension/src/node/arduino-firmware-uploader-impl.ts [hunk 1]
```diff
@@ -66,15 +70,27 @@ export class ArduinoFirmwareUploaderImpl implements ArduinoFirmwareUploader {
...
-      'flash',
-      '--fqbn',
-      firmware.board_fqbn,
-      '--address',
-      port,
-      '--module',
-      `${firmware.module}@${firmware.firmware_version}`,
-    ]);
+    this.serialService.uploadInProgress = true;
+    await this.serialService.disconnect();
```
Comment (kittaakos) [inline arduino-ide-extension/src/node/arduino-firmware-uploader-impl.ts:74]: Maybe, you can move this into the `try` block. If `disconnect` fails, the IDE won't stuck in the `uploadInProgress` state. What do you think? Just a proposal, though.

Comment (AlbyIanna) [reply arduino-ide-extension/src/node/arduino-firmware-uploader-impl.ts:74]: Makes sense! Thanks, @kittaakos!

**Review by fstasi — APPROVED** [review]

**Pull request merged**